server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -1040,7 +1053,7 @@
     /////////////////////////////////////////
 
     private static final DateTimeFormatter DATE_FORMATTER = new DateTimeFormatterBuilder()
-        .appendValue(ChronoField.YEAR, 1, 5, SignStyle.NORMAL)
+        .appendValue(ChronoField.YEAR, 1, 9, SignStyle.NORMAL)
```

Comment:
the "date_optional_time" fix for https://github.com/elastic/elasticsearch/issues/72191